js/xt.js

```diff
@@ -0,0 +1,1072 @@
+        for (let i = 0; i < ohlcvs.length; i++) {
+            const ohlcv = ohlcvs[i];
+            results.push ([
+                ohlcv['t'],
+                ohlcv['o'],
+                ohlcv['h'],
+                ohlcv['l'],
+                ohlcv['c'],
+                ohlcv['a'],
+            ]);
+        }
+        return results;
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        /**
+         * @method
+         * @name xt#fetchTicker
+         * @description fetches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        // spot
+        //
+        // {
+        //     "high": 19947.54,
+        //     "low": 18155.82,
+        //     "rate": -0.0119,
+        //     "price": 18879.54,
+        //     "moneyVol": 3018816625.5354056,
+        //     "coinVol": 159370.590698,
+        //     "ask": 18885.67,
+        //     "bid": 18879.54,
+        //     "askVol": 0.810514,
+        //     "bidVol": 41.578905,
+        //     "depthTime": 1663860050134
+        // }
+        //
+        // swap
+        //
+        // {
+        //     "high": 19315.95,
+        //     "low": 18635.03,
+        //     "rate": -0.0142,
+        //     "price": 18849.86,
+        //     "moneyVol": 3399733871.778236,
+        //     "coinVol": 178930.791303,
+        //     "ask": 18850.07,
+        //     "bid": 18846.76,
+        //     "askVol": 0.349282,
+        //     "bidVol": 9.356903,
+        //     "depthTime": 1664193429432
+        // }
+        //
+        if (market['spot']) {
+            const request = {
+                'market': market['id'],
+            };
+            const response = await this.spotPublicGetDataApiV1GetTicker (this.extend (request, params));
+            return this.parseTicker (response, market);
+        } else if (market['swap']) {
+            const request = {
+                'symbol': market['id'],
+            };
+            let method = undefined;
+            if (market['linear']) {
+                method = 'usdtPublicGetFutureMarketV1PublicQAggTicker';
+            } else if (market['inverse']) {
+                method = 'coinPublicGetFutureMarketV1PublicQAggTicker';
+            }
+            const response = await this[method] (this.extend (request, params));
+            return this.parseTicker (response, market);
+        }
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        // TODO: Add swap tickers
+        const responseSpot = await this.spotPublicGetDataApiV1GetTickers (params);
```

Comment:
when you don't have a symbol to decide whether the user wants to fetch `spot` or `swap` tickers, you should use `handleMarketTypeAndParams`. That allows the user to choose it using the defaultType or type inside params.

So the inferring logic should be:
- if the user provided at least one symbol inside `symbols` use that type
- otherwise use the defaultType or `type` in `params`